### PR TITLE
Implement Dashboard Right Sidebar

### LINKED ISSUE #557: Implement the Dashboard's right-hand sidebar
_Note: These screenshots are using an outdated Design System. The functionality remains the same today. You can see the updated version [here](https://www.figma.com/file/mxGadAHfkWH6qApebQvcdN/Group-Income-2.0?node-id=1204%3A0)._

### Problem

There is two types of information on the Dashboard:
- **Fixed information** - or information that is seldom changed, like the group members, the group description and the group mincom;
- **Mutable information** - information that can potentially change everyday, like proposals or contributions.

The fixed information needs to be visible on the Dashboard, but should not be in the way of the mutable information.
 
Captura de ecrã 2019-04-19, às 17 11 50

### Solution

Create a sidebar that keeps the fixed information at hand, but does not steal the shine from the main content. The sidebar has 3 sections:

- Minimum income;
- Members;
- About the group.

This sidebar has a width of 16,5rem and it pushes the main content to the left, as you can see in this screenshot:

Captura de ecrã 2019-04-19, às 17 18 42

Please refer to the [Dashboard mockups](https://www.figma.com/file/mxGadAHfkWH6qApebQvcdN/Group-Income-2.0?node-id=1204%3A0) for all the variations of this sidebar and further specifications.

### PR DESCRIPTION
**Pull request by sandrina-p**
Closes #557

### Dashboard sidebar default state
image

### When a member is pending (in a large group)

image

Notes:
- To see a pending member, you need to manually change `GroupMembers.vue` file. Replace all `member.pending` for `!member.pending` (cc @mmbotelho)
- I didn't understand if I should it do on the other pages too right now, so I left them as they are.

TODO:
- [x] Some misc stuff, ellipsis on the name, create "tag" component, improve DS.
- [x] "All Members modal" - Modal placeholder only.

### CODE REVIEW

**Review by mmbotelho — CHANGES_REQUESTED** [review]
I will list everything I find here - feel free to decide if this should be addressed in this PR or not.

- Disabled buttons should not have on-hover CSS changes
Captura de ecrã 2019-08-21, às 19 18 57

- Dropdown list with an icon is not as designed. You can see the [Figma component here](https://www.figma.com/file/mxGadAHfkWH6qApebQvcdN/Group-Income-2.0?node-id=2870%3A7). You can also see how the dropdown should look on the [Dashboard here](https://www.figma.com/file/mxGadAHfkWH6qApebQvcdN/Group-Income-2.0?node-id=1712%3A17).
Captura de ecrã 2019-08-21, às 19 23 44

- Mobile icon is not correct. Should be the [info icon](https://fontawesome.com/icons/info?style=solid). You can see the tablet screens [here (Figma)](https://www.figma.com/file/mxGadAHfkWH6qApebQvcdN/Group-Income-2.0?node-id=1988%3A425).
Captura de ecrã 2019-08-21, às 20 01 11

- This state should never exist (see image attached below). This is not a problem specific to this PR. This view should be the medium screen view, where both the right sidebar and the left sidebar are collapsed - [figma example here](https://www.figma.com/file/mxGadAHfkWH6qApebQvcdN/Group-Income-2.0?node-id=1988%3A864).
Captura de ecrã 2019-08-21, às 20 05 19

- Change the text in the tooltip to "We are waiting for Felix to join the group by using their unique invite link."

Captura de ecrã 2019-08-21, às 22 35 17

- This should link to the Group Settings page
Captura de ecrã 2019-08-21, às 22 42 50

**Comment by sandrina-p** [comment]
I’ll fix it all tomorrow morning @mmbotelho

**Comment by mmbotelho** [comment]
Found another one @sandrina-p 

The member dropdown should use the `button.is-icon.is-small-icon` instead of the default size
Captura de ecrã 2019-08-22, às 13 35 25
Designed version ⬆️ 

Captura de ecrã 2019-08-22, às 13 37 09
Implemented version ⬆️

**Comment by sandrina-p** [comment]
@mmbotelho added all the missing things you spot.

Also, now I'm working on  the "all members" modal.

-- EDIT --

To better explain: I just did the Modal Placeholder, not its content. I'll leave that to another PR.

**Comment by sandrina-p** [comment]
Note:
- Regarding the layout (breakpoints), I'll leave that for another PR, as well as the scroll part of the sidebar.

**Review by mmbotelho — DISMISSED** [review]

**Comment by mmbotelho** [comment]
Awesome work Sandrina! I couldn't find anything else for you to fix!

**Comment by taoeffect** [comment]
As soon as you think it's ready to be merged just remove the `WIP:` from the title 😄 

EDIT: WIP removed at @sandrina-p's request, pending @pieer's review

File: frontend/views/pages/DesignSystem.vue
Comment (taoeffect) [inline frontend/views/pages/DesignSystem.vue:438]: @sandrina-p @pieer is there a better solution, maybe with padding or margin than these `&nbsp;`s? They seem a bit hackish...

File: frontend/views/pages/DesignSystem.vue
Comment (taoeffect) [inline frontend/views/pages/DesignSystem.vue:448]: [Same question](https://github.com/okTurtles/group-income-simple/pull/630/files#r316823539)

File: frontend/views/containers/sidebar/GroupMembers.vue
Comment (taoeffect) [inline frontend/views/containers/sidebar/GroupMembers.vue]: Use `...mapGetters['memberCount']` instead (DRY)

File: frontend/views/pages/DesignSystem.vue
Comment (sandrina-p) [reply frontend/views/pages/DesignSystem.vue:438]: Oh yes, we could use paddings, but this is the DesignSystem file, like our "playground", so I didn't think it was really needed just for showcase.

File: frontend/views/containers/sidebar/GroupMembers.vue
Comment (sandrina-p) [reply frontend/views/containers/sidebar/GroupMembers.vue]: Oh, cool, done. In addition, I did the same for the getter `profilesForGroup` and realized that it was a getter not well written.

(@pieer please correct me here if I'm wrong)

Getters in Vuex work as `computed` properties in Vue. Computed properties should update its value automatically as "derived values" and should not return callbacks (another function) but `profilesForGroup` was doing that (returning a callback). That explains why doing `...mapGetters['profilesForGroup']` didn't work. So I fixed it. ([See diff here](https://github.com/okTurtles/group-income-simple/pull/630/files#diff-1ae533b411af6643b2d8345264dd5499R186))

Also, to improve consistency in naming, I renamed some getters related to groups:
- profilesForGroup -> groupMembers
- memberUsernames -> groupMembersByUsername
- memberCount -> groupMembersCount

Comment (pieer) [reply frontend/views/containers/sidebar/GroupMembers.vue]: I'm not sure to understand: why you can't use ...mapGetters['memberCount']
Or maybe the file is not updated?

File: frontend/views/containers/Overview.vue
Comment (pieer) [inline frontend/views/containers/Overview.vue]: remove the selector if not used

File: frontend/views/containers/sidebar/GroupMembers.vue
Comment (pieer) [inline frontend/views/containers/sidebar/GroupMembers.vue:119]: that's confusing. Better use direct value I reckon or leave a comment on the actual size

File: frontend/views/pages/Page.vue
Comment (pieer) [inline frontend/views/pages/Page.vue]: should be $general_2

File: frontend/views/containers/sidebar/GroupMembers.vue
Comment (sandrina-p) [reply frontend/views/containers/sidebar/GroupMembers.vue]: I assume that meanwhile, you realized that you were seeing the wrong files x)

Comment (pieer) [reply frontend/views/containers/sidebar/GroupMembers.vue]: Yes, it's all good now

File: frontend/views/containers/sidebar/GroupMembers.vue
Comment (sandrina-p) [reply frontend/views/containers/sidebar/GroupMembers.vue:119]: Based on the designs, it was neither `md` nor `sm`, it was the sum of both.

File: frontend/views/pages/Page.vue
Comment (sandrina-p) [reply frontend/views/pages/Page.vue]: fixed

File: frontend/views/containers/Overview.vue
Comment (sandrina-p) [reply frontend/views/containers/Overview.vue]: removed

File: frontend/views/containers/sidebar/GroupMembers.vue
Comment (sandrina-p) [reply frontend/views/containers/sidebar/GroupMembers.vue:119]: so that's why I did that sum (it will be compiled to the final value anyway), instead of putting a hardcoded value

**Review by pieer — APPROVED** [review]

**Pull request merged**
